source/Mission.cpp

```diff
@@ -973,6 +977,31 @@ string Mission::BlockedMessage(const PlayerInfo &player)
 
 
 
+// Get a string to show if the player is about to abort this mission.
+string Mission::AbortMessage(const PlayerInfo &player) const
+{
+	string message = "Abort mission \"" + Name() + "\"?";
+	if(!abortMessage.empty())
+	{
+		message += " " + abortMessage;
+
+		map<string, string> subs;
+		GameData::GetTextReplacements().Substitutions(subs, player.Conditions());
+		substitutions.Substitutions(subs, player.Conditions());
+		subs["<first>"] = player.FirstName();
+		subs["<last>"] = player.LastName();
+		const Ship *flagship = player.Flagship();
```

Comment:
We might also need <payment> and <fine> in substitutions, in case it will cost you to abort the mission?

Reply:
It'd be a little funky to get that working since the payment and fine could be in an on abort or on fail node.
Right now, payment and fine replacements will take from the on complete action.

Reply:
Apparent payment should work, though. I think.